packages/babel-parser/src/types.js

```diff
@@ -1183,6 +1184,11 @@ export type TsTupleType = TsTypeBase & {
   elementTypes: $ReadOnlyArray<TsType>,
 };
 
+export type TsOptionalType = TsTypeBase & {
+  type: "TSOptionalType",
+  innerType: TsType,
```

Comment:
If anyone has a better name than "innerType", I'm all ears.

Reply:
How about just `typeAnnotation`?

Reply:
Not to be too much of a choosy beggar, I hope, but `typeAnnotation` seems odd to me.  I'd consider this whole thing to be the type annotation:

```
function foo(...args: [number, string?]) {}
                      ^^^^^^^^^^^^^^^^^
```

While I'm trying to name just this part:

```
function foo(...args: [number, string?]) {}
                               ^^^^^^
```

Reply:
Both can have the property `typeAnnotation`, and they don't have conflicts.

Reply:
I'm not saying they conflict or that this property couldn't be `typeAnnotation`, but it just doesn't feel like a very clear name, either.  

Having thought on it a bit more, I think I like `optionalType`, at least more than `innerType`.